### ISSUE TITLE
YOLOv8-OBB - All the losses are 0 or nan during training and ) precision, recall metrics.

### ISSUE THREAD

**Issue by Vishawjeet-rmsl** [ultralytics/ultralytics#16290]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

I have been facing this issue for a while now where during training of the YOLOv8_-obb model, all the losses are 0 or nan (in one dataset I was getting 0, in another I'm getting nan). And the precision and recall values are also 0. It remains like that even if I train for 1200 epochs.

This is how it looks like:
![image](https://github.com/user-attachments/assets/b8c62376-bb91-4121-a49c-2aa9cea1c0b9)

I have my labels in the required format as:
![image](https://github.com/user-attachments/assets/74929453-3224-44f7-8d5f-c49e9fbffe59)

Config file is also set:
![image](https://github.com/user-attachments/assets/f5c1fc43-7bc0-4105-833f-972fd46f8712)

I have tried for over a month now, and I do not understand why this issue is arising.

### Environment

Ultralytics YOLOv8.2.77 🚀 Python-3.8.5 torch-2.0.1+cu117 CUDA:0 (NVIDIA T1200 Laptop GPU, 4096MiB)

### Minimal Reproducible Example

from ultralytics import YOLO

model = YOLO("yolov8s-obb.yaml")

res = model.train(data="myData.yaml", epochs=1200, imgsz=640, single_cls=True)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Vishawjeet-rmsl, thank you for reporting this issue and your interest in Ultralytics 🚀! This is an automated response to acknowledge your issue. An Ultralytics engineer will assist you soon.

For now, let's try to get a bit more info to help us help you better:

- If this is a 🐛 Bug Report, it would be super helpful if you can provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). This aids us in reproducing and diagnosing the problem effectively.

In the meantime, here are some resources that might be useful:
- Visit the [Docs](https://docs.ultralytics.com) for detailed instructions and common FAQs.
- Make sure your training setup follows our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Community support is also at your disposal:
- Join our [Discord](https://ultralytics.com/discord) for real-time support 🎧.
- Post on our [Discourse](https://community.ultralytics.com) for in-depth discussions.
- Explore threads on our [Subreddit](https://reddit.com/r/ultralytics).

## Install

Ensure you have pip installed the `ultralytics` package:

```bash
pip install ultralytics
```

## Environments

YOLOv8 supports various verified environments:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

The badge above shows the current status of our CI tests, which run every 24 hours across all platforms.

Stay tuned for further assistance, and thank you for your patience! 😊

**Comment by Y-T-G** [comment]
Your [GPU doesn't support AMP training](https://redd.it/1eqfl7t). You have to set `amp=False` and `half=False` during training.

**Comment by Vishawjeet-rmsl** [comment]
Thanks! 
Worked for me. Now the model is getting trained although slower.
Hopefully once the model is trained it predicts correctly.

**Comment by Y-T-G** [comment]
@Vishawjeet-rmsl Can you check if prediction works and the results are correct when you use `half=True`?

https://github.com/ultralytics/ultralytics/issues/16302#issuecomment-2352968149

Check it without using any of the fixes mentioned in that Reddit thread.

**Comment by glenn-jocher** [comment]
Great news 😃! Your original issue may now be resolved ✅ in PR #16306. 

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!
